reference/mongodb/bson/iterator/key.xml

```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<!-- $Revision$ -->
+<refentry xml:id="mongodb-bson-iterator.key" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink">
+ <refnamediv>
+  <refname>MongoDB\BSON\Iterator::key</refname>
+  <refpurpose>Returns the key of the current element</refpurpose>
+ </refnamediv>
+
+ <refsect1 role="description">
+  &reftitle.description;
+  <methodsynopsis>
+   <modifier>public</modifier> <type class="union"><type>string</type><type>int</type></type><methodname>MongoDB\BSON\Iterator::key</methodname>
+   <void/>
+  </methodsynopsis>
+ </refsect1>
+
+ <refsect1 role="parameters">
+  &reftitle.parameters;
+  &no.function.parameters;
+ </refsect1>
+
+ <refsect1 role="returnvalues">
+  &reftitle.returnvalues;
+  <para>
+   Returns the key of the current element. When iterating a BSON document, the
+   key will always be a <type>string</type>. When iterating a BSON array, the
```

Comment:
Does it make sense to refer to the class names here? I'm fine with this as-is -- just wondering if this language was intentional.

Reply:
We could use the class names here, but that's just the origin of the cursor. The idea was to differentiate the language based on the kind of structure the iterator is iterating over, not the origin of the structure (e.g. a `BSON\Document` instance).